src/scripts/main.js

```diff
@@ -1,3 +1,100 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('.field');
+const rowRemove = document.querySelector('.remove-row');
+const rowAdd = document.querySelector('.append-row');
+const columnAdd = document.querySelector('.append-column');
+const columnRemove = document.querySelector('.remove-column');
+
+const MIN_VALUE = 2;
+const MAX_VALUE = 10;
+
+document.addEventListener('click', (e) => {
+  const target = e.target.closest('.button');
+
+  if (!target) {
+    return;
+  }
+
+  changeTable(target);
+});
+
+function changeTable(elem) {
+  const addRow = elem.closest('.append-row');
+  const addColumn = elem.closest('.append-column');
+  const removeRow = elem.closest('.remove-row');
+  const removeColumn = elem.closest('.remove-column');
+
+  if (addRow) {
+    const cloneElem = table.rows[0].cloneNode(true);
+
+    table.append(cloneElem);
+
+    if (table.rows.length === MAX_VALUE) {
+      addAttribute(addRow);
+    }
+
+    check();
+  }
+
+  if (removeRow) {
+    table.rows[table.rows.length - 1].remove();
+
+    if (table.rows.length === MIN_VALUE) {
+      addAttribute(removeRow);
```

Comment:
The function `addAttribute` is called to disable the button when the minimum number of rows is reached. Ensure that this logic is correct and that the button should indeed be disabled at this point.